### ISSUE TITLE
include controller_manager/controller_manager.h gives Conflict 

### ISSUE THREAD

**Issue by RDaneelOlivav** [ros-controls/ros_control#334]
When compiling , the include of the header `controller_manager/controller_manager.h` conflicts with `hardware_interface/interface_resources.h`. They have this class and structure called `InterfaceResources`, defined in the same name space:
Does someone know how to fix this, apart from renaming one of them ( which is not a viable option )?
Is its something of the compiler that is too restrictive?

````
>> controller_base.h
namespace hardware_interface
{
  class InterfaceResources;
}

>> interface_resources.h
struct InterfaceResources
{
  InterfaceResources() {}

  InterfaceResources(const std::string& hw_iface, const std::set<std::string>& res)
    : hardware_interface(hw_iface),
      resources(res)
  {}

  /** Hardware interface type. */
  std::string hardware_interface;

  /** Resources belonging to the hardware interface. */
  std::set<std::string> resources;
};
```


```
In file included from /opt/ros/kinetic/include/controller_manager/controller_manager.h:36:
In file included from /opt/ros/kinetic/include/controller_manager/controller_spec.h:41:
/opt/ros/kinetic/include/controller_interface/controller_base.h:41:3: error: class 'InterfaceResources' was previously declared as a struct [-Werror,-Wmismatched-tags]
  class InterfaceResources;
  ^
/opt/ros/kinetic/include/hardware_interface/interface_resources.h:43:8: note: previous use is here
struct InterfaceResources
       ^
/opt/ros/kinetic/include/controller_interface/controller_base.h:41:3: note: did you mean struct here?
  class InterfaceResources;
  ^~~~~
  struct
```

**Comment by mathias-luedtke** [comment]
Thanks for reporting!

> They have this class and structure called InterfaceResources, defined in the same name space:

Actually, one is just a forward declaration of the other.
Which compiler do you use?
According to the [gcc wiki](https://gcc.gnu.org/wiki/FAQ#Wmismatched-tags) there is no `-Wmismatched-tags`. Anyway, `class` and `struct` should be treated the same in forward declarations.

> Does someone know how to fix this, apart from renaming one of them ( which is not a viable option )?

It looks like the forward declaration is superfluous (#335).

**Comment by RDaneelOlivav** [comment]
Hi,
Thanks for the fast response!
I'm using clang 3.8 as compiler and there you can find the `-Wmismatched-tags` in [Clang Warnings](https://github.com/Barro/compiler-warnings/blob/master/clang/warnings-clang-3.8.txt).
Is this (#335)  a process that you started for fixing this issue? When do you think it will be merged to the main apt-get install ros-kinetic-ros-control?

**Comment by mathias-luedtke** [comment]
> When do you think it will be merged to the main apt-get install ros-kinetic-ros-control?

This issue is not critical and not even a bug. The last kinetic release was just 2 weeks and new features will be targeted at melodic. If we merge the fix into kinetic, it problably will end up in the release in the next 3 months.

You get this error only because of your rather strict compiler flags (no problems with default settings!). You can work around it with `-Wno-error=mismatched-tags`.

**Comment by RDaneelOlivav** [comment]
Ok Thanks ;)

**Comment by mathias-luedtke** [comment]
This was resolved in #335.
